### ISSUE TITLE
App Engine Warning: TypeError: expected httplib.Message, got <type 'instance'>

### ISSUE THREAD

**Issue by speedplane** [urllib3/urllib3#1501]
While running on App Engine, I received the following error message in the logs when making HTTP requests: `TypeError: expected httplib.Message, got <type 'instance'>`. I believe this is because App Engine uses a different HTTP lib with it's own specially named instances. These instances *should* be compatible with `httplib.Message`, so the rest of the function can proceed, but the initial check will not match.

I'll put together a PR that is able to skip the `isinstance` check in `assert_header_parsing` when we're on app engine.

The complete call stack of this error is below:
```
File "project\libs\elasticsearch_six\connection\http_urllib3.py", line 172, in perform_request
    response = self.pool.urlopen(method, url, body, retries=Retry(False), headers=request_headers, **kw)
  File "project\libs\urllib3\connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "project\libs\urllib3\connectionpool.py", line 396, in _make_request
    assert_header_parsing(httplib_response.msg)
  File "project\libs\urllib3\util\response.py", line 56, in assert_header_parsing
    type(headers)))
TypeError: expected httplib.Message, got <type 'instance'>.
```

However, the error does not cause an exception, it is caught and is printed as a wanring in `_make_request`.

**Comment by speedplane** [comment]
Provided a quick fix in PR #1502.

**Comment by sethmlarson** [comment]
We haven't had this reported to us again in some time, going to close this as potentially a transient issue with App Engine? Without more information it's tough to know the root cause. (Plus App Engine isn't getting special attention in v2.x+)
